src/main/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImpl.java

```diff
@@ -412,6 +413,116 @@ protected void writeExtractCheckFile(PaymentStatus extractedStatus, PaymentProce
+                if (ObjectUtils.isNotNull(customerProfile.getSubUnitCode())) {
+                    subUnitCode = customerProfile.getSubUnitCode();
+                    LOG.info("determineCustomerProfileSubUnitCode: subUnitCode = " + subUnitCode);
+                } else {
+                    LOG.info("determineCustomerProfileSubUnitCode: No subUnitCode could be determined for customerProfile.customerShortName = " + customerProfile.getCustomerShortName());
+                }
+            } else {
+                LOG.info("determineCustomerProfileSubUnitCode: No customer profile exists for payee name = " + pdpPaymentGroup.getPayeeName());
+            }
+        } else {
+            LOG.info("determineCustomerProfileSubUnitCode: pdpPaymemtGroup.getBatch() detected as NULL.");
+        }
+        return subUnitCode;
+    }
+    
+    public int calculateMaxNumCharsFromNewNoteLine(int numCharsAlreadyExisting, int numCharsInNewNoteLine) {
+        int totalNumChars = numCharsAlreadyExisting + numCharsInNewNoteLine;
+        if (totalNumChars < CuDisbursementVoucherConstants.DV_EXTRACT_MAX_NOTE_LINE_SIZE) {
+            return numCharsInNewNoteLine;
+        } else {
+            return CuDisbursementVoucherConstants.DV_EXTRACT_MAX_NOTE_LINE_SIZE - numCharsAlreadyExisting;
+        }
+    }
+    
+    public boolean isLastCharacterSpace(String stringToCheck) {
+        return StringUtils.endsWith(stringToCheck, KFSConstants.BLANK_SPACE);
+    }
+    
+    public boolean isFirstCharacterSpace(String stringToCheck) {
+        return StringUtils.startsWith(stringToCheck, KFSConstants.BLANK_SPACE);
+    }
+    
+    public String stripLeadingSpace(String stringToCheck) {
+        return (StringUtils.isNotEmpty(stringToCheck)) ? StringUtils.removeStart(stringToCheck, KFSConstants.BLANK_SPACE) : stringToCheck;
+    }
+    
+    public String stripTrailingSpace(String stringToCheck) {
+        return (StringUtils.isNotEmpty(stringToCheck)) ? StringUtils.removeEnd(stringToCheck, KFSConstants.BLANK_SPACE) : stringToCheck;
+    }
+    
+    public int calculateMaxNumCharsFromNewNoteLine(String noteLine, String currentCheckStubDataLine) {
+        String proposedCheckStubLine = (isLastCharacterSpace(currentCheckStubDataLine) || isFirstCharacterSpace(noteLine)) ? 
+                (currentCheckStubDataLine + noteLine) : (currentCheckStubDataLine + KFSConstants.BLANK_SPACE + noteLine);
+                
+        int totalNumChars = proposedCheckStubLine.length();
+        if (totalNumChars < CuDisbursementVoucherConstants.DV_EXTRACT_MAX_NOTE_LINE_SIZE) {
+            return noteLine.length();
+        } else {
+            String wrappedText = WordUtils.wrap(proposedCheckStubLine, CuDisbursementVoucherConstants.DV_EXTRACT_MAX_NOTE_LINE_SIZE);
+            String[] constructedCheckStubLines = wrappedText.replaceAll("[\r]", "").split("\\n");
+            return noteLine.length() - constructedCheckStubLines[1].length();
```

Comment:
I would suggest using the four-arg version of WordUtils.wrap() instead. That way, you can explicitly specify "\n" as the newline, instead of deferring to the system's default and manually removing "\r" afterwards. (Unless users can enter data containing "\r" in it already?) Also, the four-arg version lets you specify whether long words should be forcibly wrapped (though doing so could affect the calculation logic as it currently stands).

In addition, will WordUtils.wrap() do any line wrapping if the proposed line is exactly the max length? If not, then you may need to tweak this to avoid an index-out-of-bounds exception on the array.

Reply:
I believe I have addressed both if these concerns with the latest code changes.